packages/solid-urql/src/createQuery.ts

```diff
@@ -0,0 +1,337 @@
+
+/** Hook to run a GraphQL query and get updated GraphQL results.
+ *
+ * @param args - a {@link CreateQueryArgs} object, to pass a `query`, `variables`, and options.
+ * @returns a {@link CreateQueryResult} tuple of a {@link CreateQueryState} result, and re-execute function.
+ *
+ * @remarks
+ * `createQuery` allows GraphQL queries to be defined and executed.
+ * Given {@link CreateQueryArgs.query}, it executes the GraphQL query with the
+ * context’s {@link Client}.
+ *
+ * The returned result updates when the `Client` has new results
+ * for the query, and changes when your input `args` change.
+ *
+ * Additionally, if the `suspense` option is enabled on the `Client`,
+ * the `createQuery` hook will suspend instead of indicating that it’s
+ * waiting for a result via {@link CreateQueryState.fetching}.
+ *
+ * @example
+ * ```tsx
+ * import { gql, createQuery } from '@urql/solid';
+ *
+ * const TodosQuery = gql`
+ *   query { todos { id, title } }
+ * `;
+ *
+ * const Todos = () => {
+ *   const [result, reExecuteQuery] = createQuery({
+ *     query: TodosQuery,
+ *   });
+ *   // ...
+ * };
+ * ```
+ */
+export const createQuery = <
+  Data = any,
+  Variables extends AnyVariables = AnyVariables,
+>(
+  args: CreateQueryArgs<Data, Variables>
+): CreateQueryResult => {
+  const client = useClient();
+  const getContext = asAccessor(args.context);
+  const getPause = asAccessor(args.pause);
+  const getRequestPolicy = asAccessor(args.requestPolicy);
+  const getVariables = asAccessor(args.variables);
+
+  const [source, setSource] = createSignal<
+    Source<OperationResult<Data, Variables>> | undefined
+  >(undefined, { equals: false });
+
+  // Combine suspense param coming from context and client with context being priority
+  const isSuspense = createMemo(() => {
+    const ctx = getContext();
+    if (ctx !== undefined && ctx.suspense !== undefined) {
+      return ctx.suspense;
+    }
+
+    return client.suspense;
+  });
+
+  const request = createRequest(args.query, getVariables() as any);
+  const context: Partial<OperationContext> = {
+    requestPolicy: getRequestPolicy(),
+    ...getContext(),
+  };
+  const operation = client.createRequestOperation('query', request, context);
+  const initialResult: CreateQueryState<Data, Variables> = {
+    operation: operation,
+    fetching: false,
+    data: undefined,
+    error: undefined,
+    extensions: undefined,
+    hasNext: false,
+    stale: false,
+  };
+
+  const [result, setResult] =
+    createStore<CreateQueryState<Data, Variables>>(initialResult);
+
+  createComputed(() => {
+    if (getPause() === true) {
+      setSource(undefined);
+      return;
+    }
+
+    const request = createRequest(args.query, getVariables() as any);
+    const context: Partial<OperationContext> = {
+      requestPolicy: getRequestPolicy(),
+      ...getContext(),
+    };
+
+    setSource(() => client.executeQuery(request, context));
+  });
+
+  createComputed(() => {
```

Comment:
As per the other comment, I can see why you're concerned that this may effect the order of effects. Essentially, `createComputed` here obviously also causes an immediate effect, which happens to be called after the one that calls `setSource`.

This isn't really a concern in `@urql/vue` due to the same reason it isn't here: call order. However, if you're concerned about this, the tests should keep this intact anyway, and in theory, we could even try to make the code bind this guarantee more tightly by making a small utility that combines the memo-like computation of `source` with `createSignal`, but I think that's unnecessary

Reply:
After some testing I think it should be fine, `createResource` internally uses `createComputed` to listen for source changes thus order should be correct